File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/HowTo/UseLowLevelCommands.cs
using System.Threading;
using System.Threading.Tasks;
using Raven.Client.Documents;
using Raven.Client.Documents.Commands;
using Raven.Client.Documents.Session;
using Raven.Client.Http;
using Sparrow.Json;

namespace Raven.Documentation.Samples.ClientApi.HowTo
{
    public class UseLowLevelCommands
    {
        public interface IInterface
        {
            #region Execute
            void Execute<TResult>(RavenCommand<TResult> command, JsonOperationContext context, SessionInfo sessionInfo = null)
            #endregion
            ;
            #region Execute_async
            Task ExecuteAsync<TResult>(RavenCommand<TResult> command, JsonOperationContext context, SessionInfo sessionInfo = null, CancellationToken token = default(CancellationToken))
            #endregion
            ;
        }

        public async Task Examples()
        {
            using (var documentStore = new DocumentStore())
            {
                #region commands_1
                using (var session = documentStore.OpenSession())
                {
                    var command = new GetDocumentsCommand("orders/1-A", null, false);
                    session.Advanced.RequestExecutor.Execute(command, session.Advanced.Context);
                    var order = (BlittableJsonReaderObject)command.Result.Results[0];
                }
                #endregion

                #region commands_1_async
                using (var session = documentStore.OpenAsyncSession())
                {
                    var command = new GetDocumentsCommand("orders/1-A", null, false);
                    await session.Advanced.RequestExecutor.ExecuteAsync(command, session.Advanced.Context);
                    var order = (BlittableJsonReaderObject)command.Result.Results[0];
                }
                #endregion

                #region commands_2
                using (var session = documentStore.OpenSession())
                {
                    var command = new DeleteDocumentCommand("employees/1-A", null);
                    session.Advanced.RequestExecutor.Execute(command, session.Advanced.Context);
                }
                #endregion

                #region commands_2_async
                using (var session = documentStore.OpenAsyncSession())
                {
                    var command = new DeleteDocumentCommand("employees/1-A", null);
                    await session.Advanced.RequestExecutor.ExecuteAsync(command, session.Advanced.Context);
                }
                #endregion
            }
        }
    }
}

File: Documentation/4.0/Samples/java/src/test/java/net/ravendb/ClientApi/HowTo/UseLowLevelCommands.java
package net.ravendb.ClientApi.HowTo;

import com.fasterxml.jackson.databind.node.ObjectNode;
import net.ravendb.client.documents.DocumentStore;
import net.ravendb.client.documents.IDocumentStore;
import net.ravendb.client.documents.commands.DeleteDocumentCommand;
import net.ravendb.client.documents.commands.GetDocumentsCommand;
import net.ravendb.client.documents.session.IDocumentSession;
import net.ravendb.client.documents.session.SessionInfo;
import net.ravendb.client.http.RavenCommand;

public class UseLowLevelCommands {

    public interface IInterface {
        //region Execute
        public <TResult> void execute(RavenCommand<TResult> command);

        public <TResult> void execute(RavenCommand<TResult> command, SessionInfo sessionInfo);
        //endregion
    }

    public void Examples() {
        try (IDocumentStore store = new DocumentStore()) {
            //region commands_1
            try (IDocumentSession session = store.openSession()) {
                GetDocumentsCommand command = new GetDocumentsCommand("orders/1-A", null, false);
                session.advanced().getRequestExecutor().execute(command);
                ObjectNode order = (ObjectNode) command.getResult().getResults().get(0);
            }
            //endregion

            //region commands_2

            try (IDocumentSession session = store.openSession()) {
                DeleteDocumentCommand command = new DeleteDocumentCommand("employees/1-A", null);
                session.advanced().getRequestExecutor().execute(command);
            }
            //endregion
        }
    }
}
